polars/polars-core/src/frame/mod.rs

```diff
@@ -437,36 +434,21 @@ impl DataFrame {
         self
     }
 
-    /// Estimates of the DataFrames columns consist of the same chunk sizes
+    /// Returns true if the chunks of the columns do not align and re-chunking should be done
```

Comment:
Could you give this one a spin? Then we don't have any allocation and still have early returns.

```rust
        let n_columns = self.columns.len();

        // outer iterator: over columns
        // inner iterator: over chunk lengths
        // flattened into a single long iterator over chunk lengths
        let mut iter= self.columns.iter().map(|s| s.chunk_lengths()).flatten();
        loop {
            // take `n_column` from the flattened iterator.
            let mut iter_row = (&mut iter).take(n_columns);
            match iter_row.next() {
                // we are done and did not have an early return, so all is equal
                None => return false,
                // check if all chunk lengths are equal to the first one
                Some(first) => if !iter_row.all(|x| first == x) {
                    return true
                }
            }
        }
```